sw/device/lib/base/mmio.h

```diff
@@ -234,6 +234,37 @@ inline void mmio_region_nonatomic_set_mask32(mmio_region_t base,
   mmio_region_write32(base, offset, value);
 }
 
+/**
+ * Masked Register field.
```

Comment:
It would be good to clarify a few things:

1. This is for fields in 32-bit fields.
2. Which end `index` counts from (when shifting).
3. Whether `mask` also gets shifted by `index` before being applied.

I think it would help to document an example, if possible (turn an instance of this struct into the value that will be written into an a 32-bit value that started as all-zeroes).

Reply:
Done.